sound/soc/codecs/cs35l56.c

```diff
@@ -772,9 +772,20 @@ static int cs35l56_component_probe(struct snd_soc_component *component)
 {
 	struct cs35l56_private *cs35l56 = snd_soc_component_get_drvdata(component);
 	struct dentry *debugfs_root = component->debugfs_root;
+	unsigned short vendor, device;
 
 	BUILD_BUG_ON(ARRAY_SIZE(cs35l56_tx_input_texts) != ARRAY_SIZE(cs35l56_tx_input_values));
 
+	if (!cs35l56->dsp.system_name &&
+	    (snd_soc_card_get_pci_ssid(component->card, &vendor, &device) == 0)) {
+		cs35l56->dsp.system_name = devm_kasprintf(cs35l56->base.dev,
+							  GFP_KERNEL,
+							  "%04x%04x",
```

Comment:
Or they might be using the same SSID because the audio hardware is identical and needs the same firmware...
Adding extra de-aliasing has the possibility of complicating things. What if a laptop comes with a choice of CPUs but has the same audio hardware, so needs the same firmware, and has the same SSID (because it's the same SSID). We'd end up with people complaining because they bought a model with a different CPU and it won't load the firmware.

I think we have to trust that the ACPI is valid and add special handling for broken cases only when that is needed.

Reply:
"ACPI is valid" is an optimistic statement ... ok to punt and see what happens.